gcc 4.6 override keyword

We still support gcc 4.6 but it doesn't support the override keyword (see http://stackoverflow.com/a/15776434), so I am getting compiler errors:
```
/ssd/deal-git/include/deal.II/lac/trilinos_sparse_matrix.h:2317:25: error: expected ‘;’ at end of member declaration
/ssd/deal-git/include/deal.II/lac/trilinos_sparse_matrix.h:2335:43: error: expected ‘;’ at end of member declaration
...
```

--- a/include/deal.II/lac/trilinos_sparse_matrix.h
+++ b/include/deal.II/lac/trilinos_sparse_matrix.h
@@ -2201,13 +2201,6 @@ namespace TrilinosWrappers
         virtual ~TrilinosPayload() {}
 
         /**
-        * Default copy assignment operator.
-        */
-        TrilinosPayload &
-        operator=(const TrilinosPayload &) = default;
-
-
-        /**
         * Returns a payload configured for transpose operations
         */
         TrilinosPayload transpose_payload () const;
@@ -2314,7 +2307,7 @@ namespace TrilinosWrappers
          * Epetra_Operator.
          */
         virtual bool
-        UseTranspose () const override;
+        UseTranspose () const;
 
         /**
          * Sets an internal flag so that all operations performed by the matrix,
@@ -2332,7 +2325,7 @@ namespace TrilinosWrappers
          * operations that may occur on different threads simultaneously.
          */
         virtual int
-        SetUseTranspose (bool UseTranspose) override;
+        SetUseTranspose (bool UseTranspose);
 
         /**
          * Apply the vmult operation on a vector @p X (of internally defined
@@ -2347,7 +2340,7 @@ namespace TrilinosWrappers
          */
         virtual int
         Apply(const VectorType &X,
-              VectorType       &Y) const override;
+              VectorType       &Y) const;
 
         /**
          * Apply the vmult inverse operation on a vector @p X (of internally
@@ -2368,7 +2361,7 @@ namespace TrilinosWrappers
          */
         virtual int
         ApplyInverse(const VectorType &Y,
-                     VectorType       &X) const override;
+                     VectorType       &X) const;
 //@}
 
         /**
@@ -2383,7 +2376,7 @@ namespace TrilinosWrappers
          * Epetra_Operator.
          */
         virtual const char *
-        Label () const override;
+        Label () const;
 
         /**
          * Returns a reference to the underlying MPI communicator for
@@ -2393,7 +2386,7 @@ namespace TrilinosWrappers
          * Epetra_Operator.
          */
         virtual const Epetra_Comm &
-        Comm () const override;
+        Comm () const;
 
         /**
          * Return the partitioning of the domain space of this matrix, i.e., the
@@ -2403,7 +2396,7 @@ namespace TrilinosWrappers
          * Epetra_Operator.
          */
         virtual const Epetra_Map &
-        OperatorDomainMap () const override;
+        OperatorDomainMap () const;
 
         /**
          * Return the partitioning of the range space of this matrix, i.e., the
@@ -2414,7 +2407,7 @@ namespace TrilinosWrappers
          * Epetra_Operator.
          */
         virtual const Epetra_Map &
-        OperatorRangeMap () const override;
+        OperatorRangeMap () const;
 //@}
 
       private:
@@ -2456,7 +2449,7 @@ namespace TrilinosWrappers
          * Epetra_Operator.
          */
         virtual bool
-        HasNormInf () const override;
+        HasNormInf () const;
 
         /**
          * Returns the infinity norm of this operator.
@@ -2466,7 +2459,7 @@ namespace TrilinosWrappers
          * Epetra_Operator.
          */
         virtual double
-        NormInf () const override;
+        NormInf () const;
       };
 
       /**

--- a/source/lac/trilinos_precondition.cc
+++ b/source/lac/trilinos_precondition.cc
@@ -73,7 +73,11 @@ namespace TrilinosWrappers
   MPI_Comm
   PreconditionBase::get_mpi_communicator () const
   {
-    return communicator.GetMpiComm();
+#ifdef DEAL_II_WITH_MPI
+    return communicator.Comm();
+#else
+    return MPI_COMM_SELF;
+#endif
   }
 
 

--- a/source/lac/trilinos_sparse_matrix.cc
+++ b/source/lac/trilinos_sparse_matrix.cc
@@ -2449,6 +2449,20 @@ namespace TrilinosWrappers
 {
   namespace internal
   {
+    namespace
+    {
+#ifndef DEAL_II_WITH_MPI
+      Epetra_Map
+      make_serial_Epetra_map (const IndexSet &serial_partitioning)
+      {
+        // See IndexSet::make_trilinos_map
+        return Epetra_Map(TrilinosWrappers::types::int_type(serial_partitioning.size()),
+                          TrilinosWrappers::types::int_type(serial_partitioning.n_elements()),
+                          0, Epetra_SerialComm());
+      }
+#endif
+    }
+
     namespace LinearOperator
     {
 
@@ -2456,9 +2470,12 @@ namespace TrilinosWrappers
         : use_transpose (false),
 #ifdef DEAL_II_WITH_MPI
           communicator (MPI_COMM_SELF),
-#endif
           domain_map (IndexSet().make_trilinos_map(communicator.Comm())),
           range_map (IndexSet().make_trilinos_map(communicator.Comm()))
+#else
+          domain_map (internal::make_serial_Epetra_map(IndexSet())),
+          range_map (internal::make_serial_Epetra_map(IndexSet()))
+#endif
       {
         vmult = [](Range &, const Domain &)
         {
@@ -2494,9 +2511,14 @@ namespace TrilinosWrappers
       TrilinosPayload::TrilinosPayload (const TrilinosWrappers::SparseMatrix &matrix_exemplar,
                                         const TrilinosWrappers::SparseMatrix &matrix)
         : use_transpose (matrix_exemplar.trilinos_matrix().UseTranspose()),
+#ifdef DEAL_II_WITH_MPI
           communicator (matrix_exemplar.get_mpi_communicator()),
           domain_map (matrix_exemplar.locally_owned_domain_indices().make_trilinos_map(communicator.Comm())),
           range_map (matrix_exemplar.locally_owned_range_indices().make_trilinos_map(communicator.Comm()))
+#else
+          domain_map (internal::make_serial_Epetra_map(matrix_exemplar.locally_owned_domain_indices())),
+          range_map (internal::make_serial_Epetra_map(matrix_exemplar.locally_owned_range_indices()))
+#endif
       {
         vmult = [&matrix_exemplar,&matrix](Range &tril_dst, const Domain &tril_src)
         {
@@ -2553,9 +2575,14 @@ namespace TrilinosWrappers
       TrilinosPayload::TrilinosPayload (const TrilinosWrappers::SparseMatrix     &matrix_exemplar,
                                         const TrilinosWrappers::PreconditionBase &preconditioner)
         : use_transpose (matrix_exemplar.trilinos_matrix().UseTranspose()),
+#ifdef DEAL_II_WITH_MPI
           communicator (matrix_exemplar.get_mpi_communicator()),
           domain_map (matrix_exemplar.locally_owned_domain_indices().make_trilinos_map(communicator.Comm())),
           range_map (matrix_exemplar.locally_owned_range_indices().make_trilinos_map(communicator.Comm()))
+#else
+          domain_map (internal::make_serial_Epetra_map(matrix_exemplar.locally_owned_domain_indices())),
+          range_map (internal::make_serial_Epetra_map(matrix_exemplar.locally_owned_range_indices()))
+#endif
       {
         vmult = [&matrix_exemplar,&preconditioner](Range &tril_dst, const Domain &tril_src)
         {
@@ -2631,9 +2658,14 @@ namespace TrilinosWrappers
       TrilinosPayload::TrilinosPayload (const TrilinosWrappers::PreconditionBase &preconditioner_exemplar,
                                         const TrilinosWrappers::PreconditionBase &preconditioner)
         : use_transpose (preconditioner_exemplar.trilinos_operator().UseTranspose()),
+#ifdef DEAL_II_WITH_MPI
           communicator (preconditioner_exemplar.get_mpi_communicator()),
           domain_map (preconditioner_exemplar.locally_owned_domain_indices().make_trilinos_map(communicator.Comm())),
           range_map (preconditioner_exemplar.locally_owned_range_indices().make_trilinos_map(communicator.Comm()))
+#else
+          domain_map (internal::make_serial_Epetra_map(preconditioner_exemplar.locally_owned_domain_indices())),
+          range_map (internal::make_serial_Epetra_map(preconditioner_exemplar.locally_owned_range_indices()))
+#endif
       {
         vmult = [&preconditioner_exemplar,&preconditioner](Range &tril_dst, const Domain &tril_src)
         {
@@ -2760,7 +2792,11 @@ namespace TrilinosWrappers
       MPI_Comm
       TrilinosPayload::get_mpi_communicator () const
       {
+#ifdef DEAL_II_WITH_MPI
         return communicator.Comm();
+#else
+        return MPI_COMM_SELF;
+#endif
       }
 
 

